### PR TITLE
Clean context on config change

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
@paulgallagher75 with it when you modify the config yaml the context will get cleared. Also `pig run` and `pig configure` will always start clean

File: pig/src/main/java/org/jboss/pnc/bacon/pig/impl/utils/HashUtils.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.jboss.resteasy.util.Hex;
+
+public class HashUtils {
+    private HashUtils() {
+    }
+
+    public static String hashDirectory(Path directory) {
+        MessageDigest sha = DigestUtils.getSha512Digest();
+        try {
+            Files.walk(directory).filter(Files::isRegularFile).sorted().forEach(path -> {
```
Comment (dwalluck) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/impl/utils/HashUtils.java:18]: I think this should be `try (FIles.walk()) {`

Comment (dwalluck) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/impl/utils/HashUtils.java:18]: I'll fix it in #402

Comment (michalszynkiewicz) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/impl/utils/HashUtils.java:18]: thanks!

**Pull request merged**